tools/odbc/odbc_utils.cpp

```diff
@@ -0,0 +1,185 @@
+}
+
+void OdbcUtils::WriteString(const string &s, SQLCHAR *out_buf, SQLSMALLINT buf_len, SQLSMALLINT *out_len) {
+	if (out_buf) {
+		snprintf((char *)out_buf, buf_len, "%s", s.c_str());
+	}
+	if (out_len) {
+		*out_len = s.size();
+	}
+}
+
+SQLRETURN OdbcUtils::SetStringValueLength(const string &val_str, SQLLEN *str_len_or_ind_ptr) {
+	if (str_len_or_ind_ptr) {
+		// it fills the required lenght from string value
+		*str_len_or_ind_ptr = val_str.size();
+		return SQL_SUCCESS;
+	}
+	// there is no length pointer
+	return SQL_ERROR;
+}
+
+SQLRETURN OdbcUtils::SetStringAndLength(vector<string> &error_messages, const string &val_str,
+                                        SQLPOINTER target_value_ptr, SQLSMALLINT buffer_length,
+                                        SQLSMALLINT *str_len_or_ind_ptr) {
+	if (!target_value_ptr) {
+		return OdbcUtils::SetStringValueLength(val_str, reinterpret_cast<SQLLEN *>(str_len_or_ind_ptr));
+	}
+
+	SQLRETURN ret = SQL_SUCCESS;
+
+	auto out_len = duckdb::MinValue(val_str.size(), (size_t)buffer_length);
+	memcpy((char *)target_value_ptr, val_str.c_str(), out_len);
+
+	if (out_len == (size_t)buffer_length) {
+		ret = SQL_SUCCESS_WITH_INFO;
+		out_len = buffer_length - 1;
+		error_messages.emplace_back("SQLGetData returned with info.");
+	}
+
+	// null terminator char
+	((char *)target_value_ptr)[out_len] = '\0';
+	if (str_len_or_ind_ptr) {
+		*str_len_or_ind_ptr = out_len;
+	}
+
+	return ret;
+}
+
+string OdbcUtils::GetStringAsIdentifier(const string &str) {
+	string str_ret;
+
+	std::regex regex_str_quoted("^(\".*\"|\'.*\')$");
+
+	if (std::regex_match(str, regex_str_quoted)) {
+		// removing quotes
+		str_ret = str_ret.substr(1, str_ret.size() - 2);
+		// removing leading and trailing blanks
+		str_ret = std::regex_replace(str, std::regex("^ +| +$|( ) +"), "$1");
+	} else {
+		// removing trailing blanks
+		str_ret = std::regex_replace(str, std::regex(" +$|( ) +"), "$1");
+		str_ret = duckdb::StringUtil::Upper(str_ret);
+	}
+
+	return str_ret;
+}
+
+string OdbcUtils::ParseStringFilter(const string &filter_name, const string &filter_value,
+                                    SQLUINTEGER sql_attr_metadata_id, const string &coalesce_str) {
+	string filter;
+	if (filter_value.empty()) {
+		if (coalesce_str.empty()) {
+			filter = "COALESCE(" + filter_name + ",'') LIKE  '%'";
+		} else {
+			filter = "COALESCE(" + filter_name + ",'" + coalesce_str + "') LIKE '" + coalesce_str + "'";
+		}
+	} else if (sql_attr_metadata_id == SQL_TRUE) {
+		filter = filter_name + "=" + OdbcUtils::GetStringAsIdentifier(filter_value);
+	} else {
+		filter = filter_name + " LIKE '" + filter_value + "'";
+	}
+	return filter;
+}
+
+string OdbcUtils::GetQueryDuckdbColumns(const string &catalog_filter, const string &schema_filter,
+                                        const string &table_filter, const string &column_filter) {
+	string sql_duckdb_columns =
+	    "SELECT "
```

Comment:
SQL statements can be made a lot more readable by [using multi-line string literals](https://stackoverflow.com/a/5460235), e.g.:

```cpp
string sql_duckdb_columns = R"(
 SELECT NULL "TABLE_CAT",
                schema_name, ...
)";
```